scripts/pylib/twister/twisterlib/coverage.py

```diff
@@ -221,11 +221,11 @@ def _generate(self, outdir, coveragelog):
         excludes = Gcovr._interleave_list("-e", self.ignores)
 
         # We want to remove tests/* and tests/ztest/test/* but save tests/ztest
-        cmd = ["gcovr", "-r", self.base_dir, "--gcov-executable",
-               str(self.gcov_tool), "-e", "tests/*"] + excludes + ["--json",
-                                                                   "-o",
-                                                                   coveragefile,
-                                                                   outdir]
+        cmd = ["gcovr", "-r", self.base_dir,
+               "--gcov-ignore-parse-errors=negative_hits.warn_once_per_file",
```

Comment:
the option to ignore warnings is added for the base_dir data collection only.
It might be considered a further improvement either to apply it only on selected parts there like `kernel/*`, or do the same also for `tests/*`.

Reply:
That looks good ;)
If for some reason, gcovr version cannot be updated to the newer one (6.0),
there may be an alternate solution - ignore coverage from `/subsys/testsuite/coverage`, i.e. adding the option for gcovr:  `-e .*/subsys/testsuite/coverage/.*`
(instead of `--gcov-ignore-parse-errors=negative_hits.warn_once_per_file`).
However, this is a worse alternate as those gcov issues may occur in other files too.